core/src/test/scala/com/yahoo/maha/core/query/MultiQueryTest.scala

```diff
@@ -24,5 +24,32 @@ class MultiQueryTest extends FunSuite with Matchers with BaseQueryGeneratorTest
         }
     }
   }
+
+  test("MultiQuery with a subsequentQuery to execute.") {
+    val model = getRequestModel(combinedQueryJson)
+    val dimQuery = getQuery(OracleEngine, getDimQueryContext(OracleEngine, model, Option("Advertiser ID")), DimOnlyQuery)
+    val dimQuery2 = getQuery(OracleEngine, getDimQueryContext(OracleEngine, model, Option("Advertiser ID")), DimOnlyQuery)
+
+    val qc = new MultiQuery(List(dimQuery, dimQuery2))
+    val irlFn = (q : Query) => new DimDrivenPartialRowList("Advertiser ID", q)
+    val result = qc.execute(queryExecutorContext, irlFn, QueryAttributes.empty, new EngineQueryStats)
+    result.rowList.foreach {
+      r =>
+        model.requestCols.map(_.alias).foreach {
+          col => assert(r.getValue(col) === s"$col-value")
+        }
+    }
+  }
+
```

Comment:
Below executes query so that the logger logs a fallback query attempt (untestable code otherwise).